File: platforms/swift/examples/core/renderer/read_texture_async.swift
import FragmentColor
let renderer = Renderer()
let texture = try await renderer.createStorageTexture(([64, 64], TextureFormat.rgba))
try texture.write(Array(repeating: 0, count: 64 * 64 * 4))

let bytes = try await renderer.readTextureAsync(texture.id())

File: platforms/python/examples/core/renderer/read_texture_async.py
from fragmentcolor import Renderer, TextureFormat
renderer = Renderer()
texture = renderer.create_storage_texture([64, 64], TextureFormat.Rgba)
texture.write([0] * (64 * 64 * 4))

# In Python, get_image() is synchronous (blocks internally via pollster).
bytes = texture.get_image()
